src/builder/modules/Generator_Builder.res

```diff
@@ -437,6 +437,48 @@ let fillBackgroundColor = (model: Model.t, fillStyle: string) => {
   }
 }
 
+let drawLine = (
+  model: Model.t,
+  (dx, dy, dw, dh): rectangle,
+  ~color: string,
+  ~width: float,
+  ~pattern: array<int>,
+  ~offset: int,
```

Comment:
Would it look OK if we don't specify an offset? Just trying to use default values unless there is a good reason to change them.

Reply:
The reason that an offset is used is so that, for instance, a fold line whose length is a power of 2 can be symmetrical, otherwise one end of the line would be opaque and the other end transparent. The offset will also help when drawing a path between multiple points, to keep the line from looking like it starts over every time it changes direction.

Reply:
I tried a few fixed offsets for a bunch of different squares:

Offset 0:

<img width="1207" alt="image" src="https://user-images.githubusercontent.com/85809598/170447570-38ea7a80-8538-422b-9303-64c5ae8c3aa3.png">

Offset 1:

<img width="1190" alt="image" src="https://user-images.githubusercontent.com/85809598/170447647-2783ec14-2475-49ee-b463-f50dbef2514f.png">

Offset 2:

<img width="1200" alt="image" src="https://user-images.githubusercontent.com/85809598/170447731-3d3fb3f3-f00b-4e9c-93ec-e85fd454489f.png">

Offset 3:

<img width="1205" alt="image" src="https://user-images.githubusercontent.com/85809598/170447860-4a6b46b0-c229-4852-a450-8a23294b471b.png">

Steve's head with offset 0:

<img width="816" alt="image" src="https://user-images.githubusercontent.com/85809598/170449442-fef72bf9-5bf6-4d5a-a7a9-251760511bb6.png">

What do you think?

Reply:
with an offset of 0, the folds on one edge will be asymmetrical as the pattern starts with an opaque pair and ends with a transparent pair. Using an offset of 3 makes the pattern start on a trailing transparent pixel and end on a leading transparent pixel for any edge length divisible by 4- most edge lengths used for most faces- and keeps it in line with the folds used for the block generator, character generators etc.